share/gpodder/extensions/tagging.py

```diff
@@ -144,76 +150,108 @@ def write_basic_tags(self, remove_before_modify, modify_tags, set_artist_to_albu
             # Not actually audio
             audio.save()
 
-    def insert_coverart(self):
+    def extract_coverart(self):
         """Implement the cover art logic in the subclass."""
         None
 
-    def get_cover_picture(self, cover):
-        """Return mutagen Picture class for the cover image.
+    def insert_coverart(self, image=None, mimetype=None):
+        """Implement the cover art logic in the subclass."""
+        None
+
+    def get_cover_picture(self, cover, mimetype):
+        """Return mutagen FLAC Picture class for the cover image.
 
         Useful for OGG and FLAC format
 
         Picture type = cover image
         see http://flac.sourceforge.net/documentation_tools_flac.html#encoding_options
         """
-        f = open(cover, mode='rb')
         p = Picture()
         p.type = 3
-        p.data = f.read()
-        p.mime = mimetypes.guess_type(cover)[0]
-        f.close()
+        p.data = cover
+        p.mime = mimetype
 
         return p
 
 
 class OggFile(AudioFile):
-    def __init__(self, filename, album, title, subtitle, genre, pubDate, cover):
-        super(OggFile, self).__init__(filename, album, title, subtitle, genre, pubDate, cover)
+    def __init__(self, filename, album, title, subtitle, genre, pubDate):
+        super(OggFile, self).__init__(filename, album, title, subtitle, genre, pubDate)
 
-    def insert_coverart(self):
+    def extract_coverart(self):
         audio = File(self.filename, easy=True)
-        p = self.get_cover_picture(self.cover)
-        audio['METADATA_BLOCK_PICTURE'] = base64.b64encode(p.write())
-        audio.save()
+        try:
+            image = base64.b64decode(str(audio['metadata_block_picture']))
+            image = Picture(image)
+        except:
```

Comment:
From [mutagen doc](https://github.com/quodlibet/mutagen/blob/62a7b3ee9f9c05321a80ba7914226189c66f28c7/docs/user/vcomment.rst#embedded-images)
I prefer to specify the exception type but I can understand it is best effort when trying to read metadata from random files from the internet. Keep the `except:` if you prefer.

```suggestion
        except (TypeError, ValueError, MutagenError):
```
(will need to add `MutagenError` to the `from mutagen import`)

Reply:
Now that I look at this, I suppose we're kind of covering both mutagen errors (and an image simply not being present) as well as PIL errors from Picture(). Hmm. Could we unconditionally catch all exceptions but log the exception type? Getting an exception will be routine for files which simply don't have an image embedded.

Reply:
Maybe something like [2af903b](https://github.com/gpodder/gpodder/pull/1701/commits/2af903b96961d18061f4c26b12e27ab36b03729d)?

Reply:
ok. But you could also check `if "metadata_block_picture" in audio:`.

Reply:
We certainly could if you prefer that approach, yeah. Shall I do that?

Reply:
it's ok like this.
Thanks